pkg/client/connection.go

```diff
@@ -37,6 +37,14 @@ type Connection interface {
 	// connection can't be reused.
 	Close() error
 
+	// PublishByteArray sends a byte array to the messaging server, which in turn
+	// sends the message to the specified destination. If the messaging server fails to
+	// receive the message for any reason, the connection will close.
+	//
+	// contentType is the MIME of the message, usually "text/plain"
+	// contentType should represent the content type of the body.
+	PublishByteArray(contentType string, body []byte, destination string) error
```

Comment:
I think that this should not be part of the contract of the connection. Publishing a message that has a body composed of bytes should, bypassing the `Publish` method should not, in my opinion.

Reply:
Sounds good to me :+1:  on it.